### PR TITLE
BUG: Mark translatable strings in module ExtensionWizard

### PR DESCRIPTION
**Pull request by MohamedAllali**
@lassoan  can you please look at this.

### CODE REVIEW

File: Modules/Loadable/Terminologies/qSlicerTerminologiesReader.cxx [hunk 1]
```diff
@@ -12,7 +12,7 @@
   See the License for the specific language governing permissions and
   limitations under the License.
 
-  This file was originally developed by Adam Rankin, PerkLab, Queen's University
+  This file was originally developed by  Adam Rankin, PerkLab, Queen's University
```
Comment (lassoan) [inline Modules/Loadable/Terminologies/qSlicerTerminologiesReader.cxx:15]: remove extra space

File: Modules/Scripted/ExtensionWizard/ExtensionWizard.py [hunk 2]
```diff
@@ -206,13 +209,13 @@ def setupTemplates(self):
                     self.templateManager.addCategoryPath(c, path)
                 except:
                     mp = (c, path)
-                    qt.qWarning("failed to add template path %r for category %r" % mp)
+                    qt.qWarning(_("failed to add template path %r for category %r") % mp)
```
Comment (lassoan) [inline Modules/Scripted/ExtensionWizard/ExtensionWizard.py:212]: fix the same way as the previous one

File: Modules/Scripted/ExtensionWizard/ExtensionWizard.py [hunk 3]
```diff
@@ -206,13 +209,13 @@ def setupTemplates(self):
...
                     qt.qWarning(traceback.format_exc())
 
     # ---------------------------------------------------------------------------
     def createExtension(self):
-        dlg = CreateComponentDialog("extension", self.parent.window())
-        dlg.setTemplates(self.templateManager.templates("extensions"))
+        dlg = CreateComponentDialog(_("extension"), self.parent.window())
+        dlg.setTemplates(self.templateManager.templates(_("extensions")))
```
Comment (lassoan) [inline Modules/Scripted/ExtensionWizard/ExtensionWizard.py:218]: I don't think is is allowed to be translated. Please test.
```suggestion
        dlg.setTemplates(self.templateManager.templates("extensions"))
```

File: Modules/Scripted/ExtensionWizard/ExtensionWizard.py [hunk 4]
```diff
@@ -286,7 +289,7 @@ def selectExtension(self, path=None):
             xp = SlicerWizard.ExtensionProject(path)
 
         except:
-            slicer.util.errorDisplay("Failed to open extension '%s'." % path, parent=self.parent.window(),
+            slicer.util.errorDisplay(_("Failed to open extension '%s'.") % path, parent=self.parent.window(),
```
Comment (lassoan) [inline Modules/Scripted/ExtensionWizard/ExtensionWizard.py:292]: Use {} placeholder as everywhere else

File: Modules/Scripted/ExtensionWizard/ExtensionWizard.py [hunk 5]
```diff
@@ -384,25 +387,26 @@ def loadModules(path=None, depth=1, modules=None, parent=None):
 
                 if len(failed):
                     if len(failed) > 1:
-                        text = "The following modules could not be registered:"
+                        text = _("The following modules could not be registered:")
```
Comment (lassoan) [inline Modules/Scripted/ExtensionWizard/ExtensionWizard.py:390]: Change it to something like `{count} modules could not be registered.` (nothing is added after the `:`, so it does not make sense to end the text with that).

File: Modules/Scripted/ExtensionWizard/ExtensionWizard.py [hunk 6]
```diff
@@ -384,25 +387,26 @@ def loadModules(path=None, depth=1, modules=None, parent=None):
...
                     else:
-                        text = "The '%s' module could not be registered:" % failed[0].key
+                        # text = _("The '%s' module could not be registered:") % failed[0].key
```
Comment (lassoan) [inline Modules/Scripted/ExtensionWizard/ExtensionWizard.py:392]: remove line that is commented out

File: Modules/Scripted/ExtensionWizard/ExtensionWizard.py [hunk 7]
```diff
@@ -412,20 +416,20 @@ def createExtensionModule(self):
             # against that just in case...
             return
 
-        dlg = CreateComponentDialog("module", self.parent.window())
-        dlg.setTemplates(self.templateManager.templates("modules"),
-                         default="scripted")
+        dlg = CreateComponentDialog(_("module"), self.parent.window())
+        dlg.setTemplates(self.templateManager.templates(_("modules")),
```
Comment (lassoan) [inline Modules/Scripted/ExtensionWizard/ExtensionWizard.py:420]: probably must not be translated, please test

File: Modules/Scripted/ExtensionWizard/ExtensionWizard.py [hunk 8]
```diff
@@ -412,20 +416,20 @@ def createExtensionModule(self):
...
+                         default=_("scripted"))
```
Comment (lassoan) [inline Modules/Scripted/ExtensionWizard/ExtensionWizard.py:421]: probably must not be translated

File: Modules/Scripted/ExtensionWizard/ExtensionWizard.py [hunk 9]
```diff
@@ -412,20 +416,20 @@ def createExtensionModule(self):
...
         dlg.showDestination = False
 
         while dlg.exec_() == qt.QDialog.Accepted:
             name = dlg.componentName
 
             try:
-                self.templateManager.copyTemplate(self.extensionLocation, "modules",
+                self.templateManager.copyTemplate(self.extensionLocation, _("modules"),
```
Comment (lassoan) [inline Modules/Scripted/ExtensionWizard/ExtensionWizard.py:428]: probably must not be translated

File: Modules/Scripted/ExtensionWizard/ExtensionWizardLib/EditExtensionMetadataDialog.py [hunk 10]
```diff
@@ -56,9 +57,9 @@ def __init__(self, parent):
 #
 # =============================================================================
 class EditExtensionMetadataDialog:
-    project = _map_property(lambda self: self.ui.nameEdit, "text")
-    category = _map_property(lambda self: self.ui.categoryEdit, "text")
-    description = _map_property(lambda self: self.ui.descriptionEdit, "plainText")
+    project = _map_property(lambda self: self.ui.nameEdit, _("text"))
```
Comment (lassoan) [inline Modules/Scripted/ExtensionWizard/ExtensionWizardLib/EditExtensionMetadataDialog.py:60]: "text" must not be translated in these lines (this is a Qt object property name)

File: Modules/Scripted/ExtensionWizard/ExtensionWizardLib/EditExtensionMetadataDialog.py [hunk 11]
```diff
@@ -113,7 +114,7 @@ def contributors(self, value):
                 item.setText(1, c[n + 1 : -1].strip())
 
             except ValueError:
-                qt.qWarning("%r: badly formatted contributor" % c)
+                qt.qWarning(_("%r: badly formatted contributor") % c)
```
Comment (lassoan) [inline Modules/Scripted/ExtensionWizard/ExtensionWizardLib/EditExtensionMetadataDialog.py:117]: use named placeholder {}

**Review by lassoan — COMMENTED** [review]
Thanks, this is quite good, there are just a couple of things to change and test - see details in the inline comments.

**Review by lassoan — APPROVED** [review]
I've implemented fixes for all my review comments. It should be good to go now.

**Comment by MohamedAllali** [comment]
Ok, done. thank you Mr. @lassoan .

**Pull request merged**